contractcourt/commit_sweep_resolver_test.go

```diff
@@ -0,0 +1,186 @@
+package contractcourt
+
+import (
+	"testing"
+	"time"
+
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/input"
+	"github.com/lightningnetwork/lnd/lnwallet"
+	"github.com/lightningnetwork/lnd/sweep"
+)
+
+type commitSweepResolverTestContext struct {
+	resolver     *commitSweepResolver
+	notifier     *mockNotifier
+	sweeper      *mockSweeper
+	resolveErr   chan error
+	nextResolver ContractResolver
+	t            *testing.T
+}
+
+func newCommitSweepResolverTestContext(t *testing.T,
+	resolution *lnwallet.CommitOutputResolution) *commitSweepResolverTestContext {
+
+	notifier := &mockNotifier{
+		epochChan: make(chan *chainntnfs.BlockEpoch),
+		spendChan: make(chan *chainntnfs.SpendDetail),
+		confChan:  make(chan *chainntnfs.TxConfirmation),
+	}
+
+	sweeper := newMockSweeper()
+
+	checkPointChan := make(chan struct{}, 1)
+
+	chainCfg := ChannelArbitratorConfig{
+		ChainArbitratorConfig: ChainArbitratorConfig{
+			Notifier: notifier,
+			Sweeper:  sweeper,
+		},
+	}
+
+	resolver := &commitSweepResolver{
+		commitResolution: *resolution,
+	}
+
+	resolver.AttachResolverKit(ResolverKit{
+		ChannelArbitratorConfig: chainCfg,
+		Checkpoint: func(_ ContractResolver) error {
+			checkPointChan <- struct{}{}
+			return nil
+		},
+	})
+
+	return &commitSweepResolverTestContext{
+		resolver: resolver,
+		notifier: notifier,
+		sweeper:  sweeper,
+		t:        t,
+	}
+}
+
+func (i *commitSweepResolverTestContext) resolve() {
+	// Start resolver.
+	i.resolveErr = make(chan error, 1)
+	go func() {
+		var err error
+		i.nextResolver, err = i.resolver.Resolve()
```

Comment:
needs mutex to access `nextResolver`? Or can send it on channel

Reply:
it seems safe to me, because `waitForResult` waits for the `resolveErr` first

Reply:
Yeah, but it is not a very nice assumption that that will always be done. Just send on a shared channel instead (I think this was done in a different PR?)?

Reply:
Fixed